File: data-sharing/src/main/java/xyz/chener/zp/datasharing/service/UserModuleService.java
package xyz.chener.zp.datasharing.service;


import com.github.pagehelper.PageInfo;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;
import xyz.chener.zp.datasharing.entity.thirdparty.UserBase;
import xyz.chener.zp.datasharing.service.impl.UserModuleServiceFallback;

import java.util.List;


@FeignClient(name = "zp-user-module",fallback = UserModuleServiceFallback.class)
public interface UserModuleService {

    @RequestMapping(value = "/api/web/getUserBaseInfo" , method = RequestMethod.GET)
    PageInfo<UserBase> getUserBaseInfo(@ModelAttribute @RequestParam("userBase") UserBase userBase
            , @RequestParam("page") Integer page
            , @RequestParam("size") Integer size);

}

File: task/src/main/kotlin/xyz/chener/zp/task/service/UserModuleService.kt
package xyz.chener.zp.task.service

import com.baomidou.mybatisplus.core.toolkit.support.SFunction
import com.github.pagehelper.PageInfo
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.*
import xyz.chener.zp.task.entity.thirdparty.OrgBase
import xyz.chener.zp.task.entity.thirdparty.UserBase
import xyz.chener.zp.task.service.impl.UserModuleServiceFallBack


@FeignClient( name = "zp-base-module", fallback = UserModuleServiceFallBack::class)
interface UserModuleService {

    @RequestMapping(value = ["/api/web/getUserBaseInfo"], method = [RequestMethod.GET])
    fun getUserBaseInfo(
        @ModelAttribute @RequestParam("userBase") userBase: UserBase?,
        @RequestParam("page") page: Int?,
        @RequestParam("size") size: Int?
    ): PageInfo<UserBase?>?

    @RequestMapping(value = ["/api/web/getUserOrgsByUsername"], method = [RequestMethod.GET])
    fun getUserOrgs(@RequestParam(value = "username") username: String?): List<OrgBase?>?

    @RequestMapping(value = ["/api/web/getUserBaseInfoByUserIds"])
    fun getUserBaseInfoByUserIds(@RequestBody userIds: List<Long?>?): List<UserBase>

}